core/src/main/java/ai/timefold/solver/core/impl/localsearch/decider/acceptor/lateacceptance/DiversifiedLateAcceptanceAcceptor.java

```diff
@@ -46,15 +46,15 @@ private void validate() {
         }
     }
 
-    @Override
     @SuppressWarnings({ "rawtypes", "unchecked" })
+    @Override
     public boolean isAccepted(LocalSearchMoveScope<Solution_> moveScope) {
         // The acceptance and replacement strategies are based on the work:
         // Diversified Late Acceptance Search by M. Namazi, C. Sanderson, M. A. H. Newton, M. M. A. Polash, and A. Sattar
         var moveScore = moveScope.getScore();
-        var current = (Score) moveScope.getStepScope().getPhaseScope().getLastCompletedStepScope().getScore();
+        var current = (InnerScore) moveScope.getStepScope().getPhaseScope().getLastCompletedStepScope().getScore();
         var previous = current;
-        var accept = moveScore.compareTo(current) == 0 || moveScore.compareTo(lateWorse) > 0;
+        var accept = moveScore.compareTo(current) == 0 || moveScore.compareTo((InnerScore) lateWorse) > 0;
```

Comment:
Please see my comment about avoiding the cast to `InnerScore`.

Reply:
Some of these are not fixable this way. But I'm doing my best to remove as many of these casts as possible.